### ISSUE TITLE
Significant Performance Gap for the visdoc tasks in VLM2Vec-v2 Qwen2-VL-2B Reproduction

### ISSUE THREAD

**Issue by FightingFighting** [TIGER-AI-Lab/VLM2Vec#149]
Hi, I just ran the code using your code and scripts. The results of the image and video task I got are similar to the paper, but there is a large gap for the task of visdoc.  I would be very appreciative if you could help with this.  The experimental setting and the results are as follows:

The result: 

Average of IMAGE  - hit@1:        0.6401 (from 36 datasets)  ---------------->  64.9  (original paper)
Average of VIDEO  - hit@1:        0.3440 (from 18 datasets)   ---------------->  34.9  (original paper)
Average of VISDOC - ndcg_linear@5:        0.6174 (from 24 datasets)   ---------------->  65.4   (original paper)
---------------------------------------------------------------------------------------------------------------------------------------
The script:

 CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
 /home/yangze/miniconda3/envs/emb/bin/torchrun --nproc_per_node=8 --master_port=22017 --max_restarts=0 \
 train.py \
 --lora \
 --lora_r 16\
 --model_name Qwen/Qwen2-VL-2B-Instruct \
 --bf16 \
 --pooling eos \
 --normalize True \
 --temperature 0.02 \
 --dataloader_num_workers 8 \
 --dataset_config scripts/train/train_alltasks.yaml \
 --run_name $EXP_NAME \
 --output_dir $EXP_DIR \
 --grad_cache True \
 --per_device_train_batch_size 128 \
 --gc_q_chunk_size 8 \
 --gc_p_chunk_size 8 \
 --interleave_batch_size 64 \
 --lr_scheduler_type linear \
 --learning_rate 5e-5 \
 --max_steps 5000 \
 --warmup_steps 100 \
 --save_steps 50 \
 --logging_steps 1 \
 --save_safetensors True \
 --remove_unused_columns False \
 --resume_from auto \
 --report_to wandb 2>&1 | tee $EXP_DIR/train.log

---------------------------------------------------------------------------------------------------------------------------------------
The detailed result for the visdoc:
ViDoRe_arxivqa  0.6914
ViDoRe_docvqa   0.3221
ViDoRe_infovqa  0.7837
ViDoRe_tabfquad 0.7942
ViDoRe_tatdqa   0.4016
ViDoRe_shiftproject     0.5897
ViDoRe_syntheticDocQA_artificial_intelligence   0.8344
ViDoRe_syntheticDocQA_energy    0.8049
ViDoRe_syntheticDocQA_government_reports        0.7635
ViDoRe_syntheticDocQA_healthcare_industry       0.8732
ViDoRe_esg_reports_human_labeled_v2     0.4655
ViDoRe_biomedical_lectures_v2_multilingual      0.3952
ViDoRe_economics_reports_v2_multilingual        0.4023
ViDoRe_esg_reports_v2_multilingual      0.4406
VisRAG_ArxivQA  0.7727
VisRAG_ChartQA  0.8282
VisRAG_MP-DocVQA        0.7676
VisRAG_SlideVQA 0.9194
VisRAG_InfoVQA  0.8871
VisRAG_PlotQA   0.6346
ViDoSeek-page   0.1987
ViDoSeek-doc    0.7566
MMLongBench-page        0.1043
MMLongBench-doc 0.3872

**Comment by zhzxlcc** [comment]
I conduct inference on the test dataset utilizing publicly available pre-trained weights, VISDOC ndcg_linear@5:  0.6483 (from 24 datasets)
BUT when I try to reproduce the result，I only GOT 0.5872 (from 24 datasets) in VISDOC test dataset, even lower than your results😭
Do u change the code, or simply run the scripts above?
How long does it take to train with 8 GPUs when using the **train_alltasks.yaml** configuration? @FightingFighting
